src/advanced/aleo-proof-system/marlin.md

```diff
@@ -6,7 +6,7 @@ The white paper that introduced [Marlin](https://eprint.iacr.org/2019/1047.pdf)
 
 - **Polynomial Commitments:** Marlin uses an efficient polynomial commitment scheme that is extractable and hiding, ensuring that no information about the committed polynomial is revealed other than what is explicitly shown in the evaluations.
 - **Algebraic Holographic Proofs (AHPs):** These are used to preprocess the circuit, transforming it into low-degree polynomials. AHPs ensure that the verifier only needs to make a small number of queries, significantly reducing verification time.
-- **Universal and Updatable SRS:** Marlin’s SRS can be used for any circuit size and can be updated securely by any party, simplifying the setup process and improving trust assumptions.
+- **Universal and Updatable SRS:** Marlin’s SRS can be used for any circuit size (bounded by available memory and computational capabilities) and can be updated securely by any party, simplifying the setup process and improving trust assumptions.
```

Comment:
It's actually bounded by a maximum degree of polynomial committed during trusted setup but not hardware capabilities.

Reply:
Some references from [Marlin](https://eprint.iacr.org/2019/1047.pdf) paper:
`The above motivates preprocessing SNARGs where the SRS is universal, which
means that the SRS supports any circuit up to a given size bound by enabling anyone, in an offline phase after
the SRS is sampled, to publicly derive a circuit-specific SRS.`

`A preprocessing argument with universal SRS is a tuple of four algorithms ARG = (G, I,P, V). The
probabilistic polynomial-time generator G, given a size bound N ∈ N, samples an SRS srs that supports
indices of size up to N.`

`8.1 Construction
We describe the construction behind Theorem 8.1, and then discuss its efficiency features.
Generator G. The generator G, on input a security parameter λ ∈ N and size bound N ∈ N, uses N to
compute a maximum degree bound D ∈ N, samples public parameters pp ← PC.Setup(1λ
, D) for the
polynomial commitment scheme PC, and then outputs srs := pp.`

Reply:
Ahhh, that makes sense...
I found this article from Lambda class that explains a bit more about the size for KZG (which marlins use and extends). Apparently it should supports circuits with a maximum size of 2^26 gates, with is the maximum degree of the polynomial used in the setup phase.
But the paper itself only mentions indeed in the introduction "any circuit up to a given size bound".

Reply:
Also Marlin doesn't have a "trusted" set up right. It's universal.

Reply:
They have a trusted setup, it produces public parameter (which is the SRS itself) that are required in polynomial commitment scheme like KGZ. I'm not sure Marlin actually uses which but assume similar because Aleo did trusted setup ceremony before. https://setup.aleo.org/

Reply:
> Ahhh, that makes sense... I found this article from Lambda class that explains a bit more about the size for KZG (which marlins use and extends). Apparently it should supports circuits with a maximum size of 2^26 gates, with is the maximum degree of the polynomial used in the setup phase. But the paper itself only mentions indeed in the introduction "any circuit up to a given size bound".

Papers are always so hard to read because they don't explain things directly but adding tonnes of definitions hahaha

Reply:
Indeed they did the trusted setup ceremony.... This was confusing...
Thanks a lot for the explanations!